OWML.ModHelper.Menus/ModInputCombinationElementMenu.cs

```diff
@@ -0,0 +1,202 @@
+            if (_cancelCommand == null)
+            {
+                _cancelCommand = new SingleAxisCommand();
+                var cancelBindingGamepad = new InputBinding(JoystickButton.Select);
+                var cancelBindingKeyboard = new InputBinding(KeyCode.Escape);
+                _cancelCommand.SetInputs(cancelBindingGamepad, cancelBindingKeyboard);
+                var commandObject = new GameObject();
+                var commandComponent = commandObject.AddComponent<ModCommandUpdater>();
+                commandComponent.Initialize(_cancelCommand);
+            }
+            var cancelPrompt = new ScreenPrompt(_cancelCommand, "Cancel");
+            var resetPrompt = new ScreenPrompt("Reset");
+            _inputMenu.SetUpPopup(message, okCommand, _cancelCommand, null, okPrompt, cancelPrompt, resetPrompt);
+            _inputMenu.GetValue<Text>("_labelText").text = message;
+        }
+
+        private void RerouteToConsole(string message)
+        {
+            Console.WriteLine("Failed to create popup for a following message:");
+            Console.WriteLine(message);
+        }
+
+        private bool OnPopupValidate()
+        {
+            var currentCombination = _inputMenu.Combination;
+            var collisions = _inputHandler.GetCollisions(currentCombination);
+            if (collisions.Count > 0 && collisions[0] != _comboName)
+            {
+                if (_twoButtonPopup == null)
+                {
+                    RerouteToConsole($"This combination collides with \"{collisions[0]}\"");
+                    return false;
+                }
+                _twoButtonPopup.EnableMenu(true);
+                _twoButtonPopup.SetUpPopup($"This combination collides with \"{collisions[0]}\"", InputLibrary.confirm2, null,
+                    new ScreenPrompt(InputLibrary.confirm2, "Ok"), new ScreenPrompt("Cancel"), true, false);
+                _twoButtonPopup.GetValue<Text>("_labelText").text = $"this combination collides with \"{collisions[0]}\"";
+                return false;
+            }
+            if (_combinationMenu == null)
+            {
+                return true;
+            }
+            var overlap = _combinationMenu.CombinationElements
+                .Any(element => element.Title == currentCombination && element != _element);
+            if (!overlap)
+            {
+                return true;
+            }
+            if (_twoButtonPopup == null)
+            {
+                RerouteToConsole("This combination already exist in this group");
+                return false;
+            }
+            _twoButtonPopup.EnableMenu(true);
+            _twoButtonPopup.SetUpPopup("This combination already exist in this group", InputLibrary.confirm2, null,
+                new ScreenPrompt(InputLibrary.confirm2, "Ok"), new ScreenPrompt("Cancel"), true, false);
+            _twoButtonPopup.GetValue<Text>("_labelText").text = $"This combination already exist in this group";
```

Comment:
This code block looks almost identical to the one above, can it be extracted to a method and reused?

Reply:
K